promql/parse.go

```diff
@@ -1074,12 +1074,15 @@ func (p *parser) checkType(node Node) (typ model.ValueType) {
 
 	case *Call:
 		nargs := len(n.Func.ArgTypes)
-		if na := nargs - n.Func.OptionalArgs; na > len(n.Args) {
+		if na := nargs - n.Func.OptionalArgs; !n.Func.Variadic && na > len(n.Args) {
 			p.errorf("expected at least %d argument(s) in call to %q, got %d", na, n.Func.Name, len(n.Args))
 		}
-		if nargs < len(n.Args) {
+		if !n.Func.Variadic && nargs < len(n.Args) {
 			p.errorf("expected at most %d argument(s) in call to %q, got %d", nargs, n.Func.Name, len(n.Args))
 		}
+		if n.Func.Variadic && nargs > len(n.Args)+1 {
+			p.errorf("expected at least %d argument(s) in call to %q, got %d", nargs, n.Func.Name, len(n.Args))
+		}
 		for i, arg := range n.Args {
```

Comment:
This will hit a panic on slice access, no?
If the arguments are variadic, it can be that `len(n.Args) > len(n.Func.ArgTypes)`. In which case for each additional element in `n.Args` we have to check against the last element of `n.Func.ArgTypes`.